React/Base/RCTConvert.h

```diff
@@ -72,10 +91,10 @@ typedef NSURL RCTFileURL;
 + (UIKeyboardType)UIKeyboardType:(id)json;
 + (UIKeyboardAppearance)UIKeyboardAppearance:(id)json;
 + (UIReturnKeyType)UIReturnKeyType:(id)json;
-+ (UIDataDetectorTypes)UIDataDetectorTypes:(id)json;
++ (RCTDataDetectorTypes)UIDataDetectorTypes:(id)json;
```

Comment:
If there's no UIDataDetectorTypes and UIBarStyle support at all on tvOS, it would probably be best to gate the definition and implementation here, and recursively fix all of the call-sites that try accessing it. That makes more sense imo than pretending we can convert it to an enum of our own?